v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/ShardFileReader.java

```diff
@@ -216,16 +216,22 @@ public List<Shard> readForwardMigrationShardingConfig(String sourceShardsFilePat
                 + "  for host "
                 + host);
       }
-
+      // If port number is passed as a number, json interprets it as a floating string like "3306.0"
+      // While we need only the integer part 3306 representing a port number, we don't want to
+      // inadvertently accept
+      // incorrect inputs like "3306.1".
+      // So we trim the decimal point followed by a string of zeros.
+      String regexForTrailingDecimal = "\\.0+$";
       Shard shard =
           new Shard(
               "",
               host,
-              dataShard.getOrDefault("port", 0).toString(),
+              dataShard.getOrDefault("port", 0).toString().replaceAll(regexForTrailingDecimal, ""),
```

Comment:
we should fail if this sort of config is passed. Unless the string is empty or can be parsed as a number, we should just fail this bit. If we go down the path of cleaning up the input, there may be a whole bunch of cases which we don't handle.

Is there any real world case you can think of where we should be handling this case ?

Reply:
We can alternatively take a string, convert it to number and cast it to int. 
However, I think we should enforce a singe format, that port should either be inputted as an int or a string.

Reply:
It's always a string in json.
when the user adds 3306 w/o quotes in the file, it gets interpreted as "3306.0" (A string with `.0` appended)
We can round this down and convert it to an integer but then we will need to also make sure we don't accept "3306.1"

So I just trimmed the ".0+" if it's present.

Reply:
A cleaner way to do this would be one of the following 
1. check the type after parsing - if it is a number, just do number.tointeger
2. use the tonumber polizy instead -https://javadoc.io/doc/com.google.code.gson/gson/2.9.0/com.google.gson/com/google/gson/ToNumberPolicy.html

Reply:
Thanks a ton! The toNubmerPolicy makes it lot more idomatic and less hacky.
I have updated the code to use the same.